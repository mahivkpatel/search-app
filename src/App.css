body {
  padding: 30px 0;
}

.App {
  text-align: center;
}

h1 {
  font-family: 'Bigelow Rules';
  font-size: 72px;
  color: #e6ecec;
}

.title {
  text-align: center;
  color: #fff;
}
