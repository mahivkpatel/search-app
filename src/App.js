import React, { Component } from 'react'
import { CardList } from './components/card-list/card-list.component'
import './App.css'
import { SearchBox } from './components/search-box/search-box.component'

class App extends Component {
  constructor() {
    super()
    this.state = { usersList: [], searchValue: '' }
  }
  componentDidMount() {
    fetch('https://jsonplaceholder.typicode.com/users')
      .then((response) => response.json())
      .then((users) => this.setState({ usersList: users }))
  }

  handleChange = (e) => {
    this.setState({ searchValue: e.target.value })
  }
  render() {
    const { usersList, searchValue } = this.state

    const filterValues = usersList.filter((users) =>
      users.name.toLowerCase().includes(searchValue.toLocaleLowerCase()),
    )
    return (
      <div className="App">
        <div className="title">
          <h1>Search User</h1>
          <SearchBox
            placeholder="search user"
            handleChange={(e) => this.setState({ searchValue: e.target.value })}
          />
        </div>
        <CardList usersList={filterValues}></CardList>
      </div>
    )
  }
}

export default App
