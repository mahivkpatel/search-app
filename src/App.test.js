import React from 'react'

import { shallow } from 'enzyme'
import App from '../src/App'
it('expect render App component', () => {
  expect(shallow(<App />)).toMatchSnapshot()
})
