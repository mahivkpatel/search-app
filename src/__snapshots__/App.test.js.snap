// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect render App component 1`] = `ShallowWrapper {}`;
